### ISSUE TITLE
DB ID Type

### ISSUE THREAD

**Issue by dgasmith** [MolSSI/QCFractal#154]
Currently throughout the code we use type `str` for Pydantic field types when we are referring to a string that can be an [ObjectId](https://docs.mongodb.com/manual/reference/method/ObjectId/). It would be good to have a custom type to validate the basic structure is correct (24 characters, no spaces). This would catch accidental "my obj name" strings.

**Comment by dgasmith** [comment]
Actually since the ID is a string representation of hex:
```
valid_chars = {"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"}
assert len(id) == 24
assert set(id) <= valid_chars
```

Takes a grand total of 500ns per id. Not bad!

**Comment by dgasmith** [comment]
Closed in #182.
